account_move_cutoff/models/account_move_line.py

```diff
@@ -0,0 +1,311 @@
+            if (
+                move_line.start_date
+                and move_line.end_date
+                and move_line.move_id.date
+                and move_line.deferred_accrual_account_id
+                and not move_line.cutoff_ids
+                and move_line.has_deferred_dates()
+            ):
+                move_line.is_deferrable_line = True
+            else:
+                move_line.is_deferrable_line = False
+
+    def has_deferred_dates(self):
+        """Compute the account move line should be split:
+        has service in future periods
+        """
+        self.ensure_one()
+        if self._period_from_date(self.end_date) > self._period_from_date(
+            self.move_id.date
+        ):
+            return True
+        return False
+
+    @api.model
+    def _get_first_day_from_period(self, period):
+        return period
+
+    @api.model
+    def _get_last_day_from_period(self, period):
+        return self._last_day_of_month(period)
+
+    @api.model
+    def _line_amounts_on_proper_periods(self, line_amounts, periods):
+        extra_amount_first_period = 0
+        extra_amount_last_period = 0
+        amounts = defaultdict(lambda: 0)
+        date_first_period = self._get_first_day_from_period(periods[0])
+        date_end_period = self._get_last_day_from_period(periods[-1])
+        for period, line_period_amount in line_amounts.items():
+            if self._get_last_day_from_period(period) < date_first_period:
+                extra_amount_first_period += line_period_amount
+                continue
+            if self._get_first_day_from_period(period) > date_end_period:
+                extra_amount_last_period += line_period_amount
+                continue
+            amounts[period] = line_period_amount
+        amounts[periods[0]] += extra_amount_first_period
+        amounts[periods[-1]] += extra_amount_last_period
+        return amounts
+
+    def _round_amounts(self, line_periods):
+        """This method is used to round values and avoid
+        rounding side effects
+        """
+        amounts = {}
+        sum_from_next_periods = 0
+        first_period = None
+        for index, item in enumerate(line_periods.items()):
+            period, amount = item
+            amounts[period] = self.currency_id.round(amount)
+            if index > 0:
+                sum_from_next_periods += amounts[period]
+            else:
+                first_period = period
+        # avoid rounding difference
+        if first_period:
+            amounts[first_period] = self.currency_id.round(
+                abs(self.balance) - sum_from_next_periods
+            )
+        return amounts
+
+    def _get_amounts_per_periods(self, periods):
+        """dispatch to the proper method"""
+        self.ensure_one()
+        line_amounts = {}
+        if self.cutoff_method == "equal":
+            line_amounts = self._get_amounts_per_periods_equal()
+        else:  # monthly_prorata_temporis
```

Comment:
`elif` with exception if cutoff_method is not recognized ? this seems dangerous otherwise

Reply:
`cutoff_method` is a field selection with 2 value today, ORM ensure possible value before to be able to call this code I think. So it will be hard to test exception case :thinking: